### PR TITLE
various test fixes

### PR DESCRIPTION
**Pull request by justinstoller**
(: 

This is for SERVER-2304 which is implemented in 2d12e97. In addition I did a couple small test clean ups and SERVER-2303 since it was a two line change. Happy to move those into their own PRs if folks prefer.

### CODE REVIEW

File: spec/puppetserver/ca/action/generate_spec.rb [hunk 1]
```diff
@@ -154,4 +162,74 @@
...
+    Dir.mktmpdir do |tmpdir|
+      with_files_in tmpdir do |bundle, key, chain, conf|
+        private_path = File.join(tmpdir, 'ssl', 'private_keys', 'foocert.pem')
+        public_path = File.join(tmpdir, 'ssl', 'public_keys', 'foocert.pem')
+        cert_path = File.join(tmpdir, 'ssl', 'certs', 'foocert.pem')
+
+        FileUtils.mkdir_p(File.dirname(private_path))
+        FileUtils.mkdir_p(File.dirname(public_path))
+
+        pkey = OpenSSL::PKey::RSA.new(1024)
```
[the remark this answers is not in the document]
Comment (justinstoller) [reply spec/puppetserver/ca/action/generate_spec.rb]: Good catch, I looked for all things `keylength` but missed the direct creation of keys without referencing keylength.

File: lib/puppetserver/ca/action/generate.rb [hunk 2]
```diff
@@ -106,7 +106,9 @@ def generate_pki(settings, signing_digest)
             [settings[:cakey], int_key],
           ]
 
-          errors = FileSystem.check_for_existing_files(public_files.map(&:first) + private_files.map(&:first))
+          files_to_check = public_files + private_files
+          files_to_check = files_to_check.map(&:first) - [settings[:hostpubkey], settings[:hostprivkey]]
```
[the remark this answers is not in the document]
Comment (justinstoller) [reply lib/puppetserver/ca/action/generate.rb]: Yeah, great point. I've added comments in both the actions and the Host class.

**Comment by justinstoller** [comment]
addressed comments

**Comment by Magisus** [comment]
Manually tested this by attempting an agent run with no server, then used the generate command. The CA and master cert were generated correctly with no error messaging.

**Pull request merged**